src/Test/ORMDatabaseResetter.php

```diff
@@ -101,9 +103,21 @@ private function dropSchema(): void
     private function dropAndResetDatabase(): void
     {
         foreach ($this->connectionsToReset() as $connection) {
+            if ($this->registry->getConnection($connection)->getDatabasePlatform() instanceof PostgreSQLPlatform) {
```

Comment:
working for me with `PostgreSQL100Platform`.

the class is deprecated but the newer class isn't working yet.
```
Deprecated:
This class will be merged with PostgreSQLPlatform in 4.0 because support for Postgres releases prior to 10.0 will be dropped.
```

Suggest: inplement with an OR clause ..

@nikophil

Reply:
hi @benblub 

thanks for having tested that!

> Suggest: inplement with an OR clause ..

not sure to understand what you mean?

Reply:
```php
if ($this->registry->getConnection($connection)->getDatabasePlatform() instanceof PostgreSQLPlatform || $this->registry->getConnection($connection)->getDatabasePlatform() instanceof PostgreSQL100Platform)
```

@nikophil

Reply:
ok then just replace PostgreSQLPlatform with PostgreSQL100Platform. In my test PostgreSQL100Platform is working and PostgreSQLPlatform not.

Reply:
very strang it does not enter in the second if since `PostgreSQL100Platform` extends `PostgreSQLPlatform` :thinking: 

actually even when I enter the `if` and the query is played, it seems it does not kill other connections and the drop still fails... I really don't know why

Reply:
For me: if its enter the if, the other connection is killed.

Reply:
<img width="1215" alt="Bildschirmfoto 2023-05-10 um 15 40 14" src="https://github.com/zenstruck/foundry/assets/1756031/f415557a-1955-49bc-b32b-50148901e3e0">


<img width="1075" alt="Bildschirmfoto 2023-05-10 um 15 41 02" src="https://github.com/zenstruck/foundry/assets/1756031/ef325a8e-0682-40db-b3f8-e679ebffd284">

Reply:
ok i tested now a little bit more. 

instance of is working on all 3 classes like expected.. not sure this didn't entered the if last days 🤔 
PostgreSQL100Platform
PostgreSQL94Platform
PostgreSQLPlatform

**so killing the connection is working for me with all 3 classes**

after go again in the postgreSQL docs
```
however only superusers can terminate superuser backends.
```

**on your test you are maybe not a superuser???**

So basically we need to figure out if it was possible to drop the other connections?
if I'm not `SELECT rolsuper FROM pg_roles WHERE rolname = current_user;` a superuser we do not need to try go further.

It's possible to check after if connections was dropped `SELECT COUNT(*) AS active_connections FROM pg_stat_activity WHERE state IS NOT NULL` or `SELECT COUNT(*) AS active_connections FROM pg_stat_activity WHERE state = 'active'`

also we can setup a timeout and pg_terminate_backend will then return a false.

> pg_terminate_backend ( pid integer, timeout bigint DEFAULT 0 ) → boolean
> Terminates the session whose backend process has the specified process ID. This is also allowed if the calling role is a member of the role whose backend is being terminated or the calling role has privileges of pg_signal_backend, however only superusers can terminate superuser backends.
> If timeout is not specified or zero, this function returns true whether the process actually terminates or not, indicating only that the sending of the signal was successful. If the timeout is specified (in milliseconds) and greater than zero, the function waits until the process is actually terminated or until the given time has passed. If the process is terminated, the function returns true. On timeout, a warning is emitted and false is returned.

https://www.postgresql.org/docs/15/functions-admin.html

@nikophil

Reply:
If we add posgreSQL to docker we can write tests.. not sure its worth at all 🙈

Reply:
@nikophil and I have talked about doing that. Since postgres is the default for Symfony, I think it makes sense.

Reply:
I'm gonna change the image from mysql to postgres in this PR ;)
I've found why the connexion was not killed: I was using a persistant connexion 🤦‍♂️

Reply:
Maybe we should add the db platform to the matrix? I'm fine keeping it limited to 2 MySQL jobs: one with dama and one without. Just for a sanity check.

Reply:
Yes, that was planned 🙂

Reply:
I'm struggling with the CI :disappointed: 

I used to have this error see [this job](https://github.com/zenstruck/foundry/actions/runs/4967955407/jobs/8890273755)
```
SQLSTATE[HY000]: General error: 7 FATAL:  terminating connection due to administrator command
server closed the connection unexpectedly
```

then I added a condition where I say "dont kill other connections if in the CI"
but I still get [some errors](https://github.com/zenstruck/foundry/actions/runs/4968012115/jobs/8890366127) in cases without dama:
```
1) Zenstruck\Foundry\Tests\Functional\WithMigrationTest::it_generates_a_valid_schema
RuntimeException: Error running "doctrine:database:drop": Could not drop database "zenstruck_foundry" for connection named default
An exception occurred while executing a query: SQLSTATE[55006]: Object in use: 7 ERROR:  database "zenstruck_foundry" is being accessed by other users
DETAIL:  There is 1 other session using the database.
```

I would appreciate some ideas, otherwise, I think we won't use postgres in the CI :innocent:

Reply:
Is this the same problem as I had in https://github.com/zenstruck/foundry/pull/73?

Reply:
don't know, the logs have expired

Reply:
I really don't understand what's going on: https://github.com/zenstruck/foundry/actions/runs/5000161439/jobs/8957237306

I changed the CI to have only ONE job with postgres, and the db is named `zenstruck_foundry_${{ matrix.use-dama }}_${{ matrix.orm-db }}`, ie: `zenstruck_foundry_1_postgres` so it is unique in the matrix...

what I've noticed is also that the CI with postgres works without dama... :thinking:

Reply:
> what I've noticed is also that the CI with postgres works without dama...

Ok, that was exactly the issue in #73. Worked without dama but not with.

Reply:
I managed to make dama work with postgres in GH CI: we must add `?serverVersion=15` in the dsn

Reply:
ok then, the CI is green with postgres as the main db engine!

this was... painful :sweat_smile: